### DISCUSSION TITLE
Using Intersections

### DISCUSSION THREAD

**Discussion by AmmarZeer** [xyflow/xyflow#2627] [Q&A] [answered]
I am trying to use `getIntersectingNodes`, but when am destructuring it from useReactFlow I keep getting this error: 

- Property 'getIntersectingNodes' does not exist on type 'ReactFlowInstance<any, any>'.ts(2339)

Any idea why?

**Comment by bcakmakoglu** [comment, marked as answer]
What version of react flow are you using?
The intersection helpers have been added with `11.2.0` (see the [CHANGELOG](https://github.com/wbkd/react-flow/blob/main/packages/core/CHANGELOG.md#1120))

**Reply by AmmarZeer** [reply]
Yes that's correct, I've been using react-flow-renderer package instead of reactflow.

But another error that I face now is:
- Uncaught Error: Seems like you have not used zustand provider as an ancestor

This should be fixed by wrapping my component with `<ReactFlowProvider>`, but wrapping it didn't fix the issue, is there another fix for this issue?

**Reply by bcakmakoglu** [reply]
Are you using `ReactFlowProvider` inside the same component you call `useReactFlow` in?
You need to wrap it from the parent component, for example

```html
    <div className="zoompanflow">
      <ReactFlowProvider>
   		<Child />  
      </ReactFlowProvider>
    </div>
```

```js
// Child.js
const ChildComponent = () => {
  const [nodes, setNodes, onNodesChange] = useNodesState(initialNodes);
  const [edges, setEdges, onEdgesChange] = useEdgesState(initialEdges);
  const onConnect = useCallback((params) => setEdges((els) => addEdge(params, els)), []);
  const { getIntersectingNodes } = useReactFlow()

  return (
        <ReactFlow
          nodes={nodes}
          edges={edges}
          onNodesChange={onNodesChange}
          onEdgesChange={onEdgesChange}
          onConnect={onConnect}
          fitView
        />
  );
};

**Reply by AmmarZeer** [reply]
yes I am wrapping it from the parent component, but the issue still occurs.

**Reply by bcakmakoglu** [reply]
Do you have some code to share? would make it easier to find out what's causing the issue 😄 
According to the [troubleshooting](https://reactflow.dev/docs/guides/troubleshooting/) page, you need to do it as I described but if you say that's already what you do, maybe there's something else wrong

**Reply by AmmarZeer** [reply]
sure I can share the important part of the code 
PS: one of the imports has "react-flow-renderer" as i was using that package before, but changing it to "reactflow" didn't solve it
so I am wrapping it inside the router component like so:
```
       <Route
          path={"/canvas"}
          element={
            <ReactFlowProvider>
              <Canvas />
            </ReactFlowProvider>
          }
        />
```
        
Inside the component called canvas:
       
```
 import ReactFlow, {
  addEdge,
  applyEdgeChanges,
  applyNodeChanges,
  Background,
  BackgroundVariant,
  ConnectionMode,
  useViewport,
} from "react-flow-renderer";
import { useReactFlow } from "reactflow";

export default function Canvas() {
  const [nodes, setNodes] = useState([]);
  const [edges, setEdges] = useState([]);
  const { getIntersectingNodes } = useReactFlow();
  
  const onNodeDragStop = useCallback((_, node) => {
    console.log( _);
    console.log( node);
    console.log(getIntersectingNodes(node));
  }, []);
      return (
    <div
      className={styles.canvasContainer}
      style={{ height: `${height}px` }}
      ref={reactFlowWrapper}
    >
      <ReactFlow
        nodes={nodes}
        edges={edges}
        onNodeDragStop={onNodeDragStop}
        />
      </div>
        )}
```

**Reply by bcakmakoglu** [reply]
You shouldn't mix the packages. `react-flow-renderer` is deprecated and you should switch to the new package alltogether.

**Reply by AmmarZeer** [reply]
Thank you for your help, you were right, i removed the old package and installed the new one, the zustand error was resolved.
guess i'll have to make my own intersection detection with the old package, was really eager to use this new feature though.
<img width="1392" alt="image" src="https://user-images.githubusercontent.com/93482734/205350599-b31dfd04-65ca-457c-8dea-283276ca9e58.png">

**Reply by moklick** [reply]
Why can't you use the latest version? The error you showed happens when you have multiple version of react or ReactFlow installed. Try to remove your node_modules and install again and make sure there is only one version of react and react flow.
